### PR TITLE
Add option to set where new tracks are placed

### PR DESCRIPTION
**Pull request by jonata**
Resolves: #2392, [Forum discussion](https://forum.audacityteam.org/t/why-are-new-tracks-always-added-to-the-end-you-should-be-able-to-add-them-after-the-selected-track/52910/23)

When the user duplicates a selection or simply creates a new track, it is always added to the end (bottom) of the project. With this PR, we can set the default behavior/placement when new tracks are added.

- [X] I signed [CLA](https://www.audacityteam.org/cla/)
- [X] The title of the pull request describes an issue it addresses
- [X] If changes are extensive, then there is a sequence of easily reviewable commits
- [X] Each commit's message describes its purpose and effects
- [X] There are no behavior changes unnecessary for the stated purpose of the PR

Recommended:
- [X] Each commit compiles and runs on my machine without known undesirable changes of behavior

### CODE REVIEW

**Comment by LWinterberg** [comment]
Please move the preference setting into the Tracks page, as it'd have a buddy in there:

image

**Comment by jonata** [comment]
Sure thing. Do you already have this name feature implemented? Want me to implement in this PR?

**Comment by LWinterberg** [comment]
The name feature currently lives in the "display" section on the tracks page, you'll just need to move it downstairs.

**Comment by jonata** [comment]
Yes, 🤦 .

**Comment by petersampsonaudacity** [comment]
Testing on W10 with @jonata 's branch build:   audacity-win-3.6.0-alpha-20240522+81bb202-x64

@LWinterberg 

Whilst the placement settings work properly for adding new stereo and mono tracks and for duplications - when adding a Label track this is always added to the bottom of the project, regardless of the settings.

Users who are advanced users of Sync-lock will want their label tracks in various places in their project as label tracks are used to define sync-locked track groups - so adding or duplicating a label track should also follow the placement setting the user makes in prefs.

_It is OK that adding a Time Track always goes to the top of the project - that has always been the case, it was that way by design at the outset when they were first introduced._

**Comment by jonata** [comment]
Hey @petersampsonaudacity , thanks for the feedback. So, do you suggest that we can have two settings, one for "wave" tracks and another for "labels" track? I'm asking as someone who doesn't uses label tracks so often, so maybe a different setting for different type of tracks would be useful...?

**Comment by petersampsonaudacity** [comment]
> Hey @petersampsonaudacity , thanks for the feedback. So, do you suggest that we can have two settings, one for "wave" tracks and another for "labels" track? I'm asking as someone who doesn't use label tracks so often, so maybe a different setting for different type of tracks would be useful...?

Hi @jonata  - You'd have to discuss that with @LWinterberg really. it would be his call for if we had an extra setting for Label tracks _(I'm just a part-time pro-bono external tester, and documenter, these days, not a Muse employee)._

But note that further testing shows that Split New and Record do _**NOT**_ observe the placement setting for new audio tracks either - and I think they should,  Same for the Analyze tools that produce label tracks as their output.

**Comment by petersampsonaudacity** [comment]
> Do you already have this name feature implemented? Want me to implement in this PR?

AFAICT that seem to be properly working already @jonata

**Comment by jonata** [comment]
> But note that further testing shows that Split New and Record do _**NOT**_ observe the placement setting for new audio tracks either - and I think they should,
> 
> Same for the Analyze tools that produce label tracks as their output.

I agree. We should define in which actions this would be useful.

> AFAICT that seem to be properly working already jonata

It is, my fault not noticing it :)

File: src/tracks/playabletrack/wavetrack/ui/WaveTrackMenuItems.cpp [hunk 1]
```diff
@@ -59,16 +72,31 @@ void OnNewStereoTrack(const CommandContext &context)
...
+      tracks.Insert(*tracks.begin(), track);
+   } else if (newTrackPlacement == wxT("AboveCurrentTrack")) {
+      auto focusedTrack = TrackFocus::Get(project).Get();
+      tracks.Insert(focusedTrack, track);
+   } else if (newTrackPlacement == wxT("BelowCurrentTrack")) {
+      auto focusedTrack = TrackFocus::Get(project).Get();
+      tracks.Insert(tracks.GetNext(*focusedTrack), track);
+   } else {
+      tracks.Add(track);
+   }
```
Comment (grliszas14) [inline src/tracks/playabletrack/wavetrack/ui/WaveTrackMenuItems.cpp:89]: Similar block of code repeats 3 times - it can be one function used across these 3 places

File: libraries/lib-track/Track.h [hunk 2]
```diff
@@ -1117,6 +1117,9 @@ class TRACK_API TrackList final
    //! Remove and return the first track
    Track::Holder DetachFirst();
 
+   Track *GetPrev(Track &, bool linked = false) const;
+   Track *GetNext(Track &, bool linked = false) const;
```
Comment (grliszas14) [inline libraries/lib-track/Track.h:1121]: that looks like an unnecessary change

Comment (jonata) [reply libraries/lib-track/Track.h:1121]: It's needed, yes.

File: src/tracks/playabletrack/wavetrack/ui/WaveTrackMenuItems.cpp (hunk 1 above)
Comment (jonata) [reply src/tracks/playabletrack/wavetrack/ui/WaveTrackMenuItems.cpp:89]: @grliszas14 , do you have a suggestion where this function could be placed? I was thinking on `WaveTrackUtilities`, or even on `Tracks.cpp` but here I was not able to import `TrackFocus`.

**Comment by petersampsonaudacity** [comment]
> I think that when creating new tracks, the focused track should be used, and for Duplicate, the selected track or tracks should be used.

@DavidBailes 
I am somewhat minded to agree

a) In the simple use case the user has a single track selected which gets the focus because of its selection - so focus and selection are the same

b) it gets more complex in the use case with multiple tracks selected when it is likely the the last track selected will get the focus

But my big concerns are:
1) most users overlook Focus and have no idea what it is
2) the Focus border color in the new Light theme is quite hard to spot - see #6702

**Comment by LWinterberg** [comment]
Ah, sorry, I dropped the ball on this one. Same as the others - we just [switched master to 4.0.0](https://forum.audacityteam.org/t/master-has-been-switched-to-4-0-0-development/120444) which makes this PR incompatible. Audacity 4 will rework track importing and placing rules (specifically, when dragging and dropping it'll be placed at your mouse cursor).
